collections/nemo_nlp/nemo_nlp/data/datasets/utils.py

```diff
@@ -778,21 +787,210 @@ def process_dialogflow(
         slot_labels)
 
     train_queries, train_slots, test_queries, test_slots = \
-        partition_df_data(intent_queries, slot_tags, split=dev_split)
+        partition_data(intent_queries, slot_tags, split=dev_split)
 
-    write_df_files(train_queries, f'{outfold}/train.tsv')
-    write_df_files(train_slots, f'{outfold}/train_slots.tsv')
+    write_files(train_queries, f'{outfold}/train.tsv')
+    write_files(train_slots, f'{outfold}/train_slots.tsv')
 
-    write_df_files(test_queries, f'{outfold}/test.tsv')
-    write_df_files(test_slots, f'{outfold}/test_slots.tsv')
+    write_files(test_queries, f'{outfold}/test.tsv')
+    write_files(test_slots, f'{outfold}/test_slots.tsv')
 
-    write_df_files(slot_labels, f'{outfold}/dict.slots.csv')
-    write_df_files(intent_names, f'{outfold}/dict.intents.csv')
+    write_files(slot_labels, f'{outfold}/dict.slots.csv')
+    write_files(intent_names, f'{outfold}/dict.intents.csv')
 
     return outfold
 
 
-def write_df_files(data, outfile):
+def readCSV(file_path):
+    rows = []
+    with open(file_path, 'r') as csvfile:
+        readCSV = csv.reader(csvfile, delimiter=',')
+        for row in readCSV:
+            rows.append(row)
+    return rows
+
+
+def get_intents_mturk(utterances, outfold):
+
+    intent_names = {}
+    intent_count = 0
+
+    agreedallDict = {}
+
+    print('Printing all intent_labels')
+    intent_dict = f'{outfold}/dict.intents.csv'
+    if os.path.exists(intent_dict):
+        with open(intent_dict, 'r') as f:
+            for intent_name in f.readlines():
+                intent_names[intent_name.strip()] = str(intent_count)
+                intent_count += 1
+    print(intent_names)
+
+    for i, utterance in enumerate(utterances[1:]):
+
+        if utterance[1] not in agreedallDict:
+            agreedallDict[utterance[0]] = utterance[1]
+
+        if utterance[1] not in intent_names:
+            intent_names[utterance[1]] = str(intent_count)
```

Comment:
i'm not sure why intent_count/count has to be converted to str

Reply:
intent_count will be the slot number value in the slot tags which will be in string format. So it would need to be converter either here or where it's getting concatenated with the slot tag string.

Reply:
it might make more sense to store things in numbers and convert to strings only when necessary. numbers are easier/faster to manipulate and require less memory to store.